### ISSUE TITLE
Joining problems when mixing versioned and unversioned models

### ISSUE THREAD

**Issue by raphaelm** [swisscom/cleanerversion#57]
If you have two models
```python
class OrderPosition(Versionable):
    …

class OrderPosition(models.Model):
    order = VersionedForeignKey(
        Order,
        verbose_name=_("Order")
    )
    status = models.CharField(…)
```
a lookup like
```python
 OrderPosition.objects.filter(order__status=Order.STATUS_PAID)
```
will fail with ``ValueError("joined_alias not set")`` (triggered in versions/models.py:271). When I change the OrderPosition model to be versionable as well (which I now did, do avoid further confusion), all works well. However, for the future, one might want to fix this, if it is possible without too much effort.

**Comment by maennel** [comment]
Hi @raphaelm, I think, this should be fixed by now. However, I'm not entirely sure. I'll take your test case and verify...

**Comment by peterfarrell** [comment]
@boydjohnson could you confirm this is fixed?  I vaguely recall having a similar issue a few months ago.

**Comment by boydjohnson** [comment]
I'm having the same problem on a test case of 
'''class Order(Versionable):
         name = models.CharField...
 
 class OrderPosition(models.Model):
       name = models.CharField...
       order = VersionedForeignKey('Order')'''

and then 
''' OrderPosition.objects.filter(order__name = 'Anything')'''

fails with ValueError("joined_alias not set") from CleanerVersion/versions/models.py", line 386, in as_sql

**Comment by peterfarrell** [comment]
@boydjohnson thanks for confirming.  Does it work if you use `as_of()` as part of the expression?

https://github.com/swisscom/cleanerversion/blob/master/versions/models.py#L382-L386

**Comment by boydjohnson** [comment]
@peterfarrell Since OrderPosition is an unversioned class it has no as_of.

Also,
 I can get the query to work from Order (which is Versionable) to OrderPosition (which isn't).  

'''Order.objects.filter(position__name = 'Anything')''' works when the related_query_name is set to position.

**Comment by peterfarrell** [comment]
@boydjohnson  and I have figured out a suggested fix for this -- really simple and it doesn't break any of the current test suite.  The assertion checking on joined_alias is a bit over zealous in this case.  Watch for a PR from Boyd.

**Comment by maennel** [comment]
@peterfarrell and @boydjohnson Agree. I've taken over your PR and refined the assertion check. This issue can be closed after PR #103 has been merged.

**Comment by maennel** [comment]
Merged.
